client/http.go

```diff
@@ -0,0 +1,88 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/drand/drand/beacon"
+	"github.com/drand/drand/key"
+)
+
+// NewHTTPClient creates a new client pointing to an HTTP endpoint
+func NewHTTPClient(url string, groupHash []byte, client *http.Client) (Client, error) {
+	c := &httpClient{
+		root:   url,
+		client: client,
+	}
+	// fetch the `Group` to validate connectivity.
+	groupResp, err := client.Get(fmt.Sprintf("%s/group", url))
```

Comment:
Please don't have side-effects in constructors. This makes it difficult for consumers (including various consumer integration and functional-test set-ups) to control when things happen. E.g. it makes it impossible to even instantiate this object for integration tests without being exposed to network flakiness.

Split into a constuctor and a `FetchGroupInfo` method. It's fine and handy also to have a helper method that does the construction and configuration at once, but don't make that the *only* entry point.